notebooks/WFC3/point_spread_function/psf_utilities.py

```diff
@@ -0,0 +1,402 @@
+    stellar and background photometric apertures that were calculated with 
+    the photutils CircularAperture and CircularAnnulus functions, respectively.
+    '''
+    
+    # Create a figure with the data, identified stars, and photometry apertures.
+    my_figure_size, my_fontsize = setup_matplotlib('notebook', 1.2)
+    figure, axes = matplotlib.pyplot.subplots(1, 3, figsize=(11.0, 11.0/3))
+    # current_width = 1.2 * matplotlib.pyplot.rcParams["lines.linewidth"]
+    for ax in [0, 1, 2]:
+        axes[ax].imshow(data, origin='lower', aspect='equal', interpolation='nearest', norm=simple_norm(data, 'sqrt', percent=99.8))
+    apertures_stellar.plot(ax=axes[1], color='lime', alpha=1.0)
+    apertures_stellar.plot(ax=axes[2], color='lime', alpha=1.0)
+    apertures_annulus.plot(color='white')
+    axes[0].set_title('Original Data')
+    axes[1].set_title('Identified Stars')
+    axes[2].set_title('Photometry Apertures')
+    axes[0].set_ylabel('Y (pixels)')
+    axes[1].set_xlabel('X (pixels)')
+    for ax in [0, 1, 2]: 
+        axes[ax].set_xlim([xcenter-cutout_size/2, xcenter+cutout_size/2])
+        axes[ax].set_ylim([ycenter-cutout_size/2, ycenter+cutout_size/2])
+    matplotlib.pyplot.tight_layout()
+
+    return figure
+
+
+def plot_psf_results(sci_data, resid, xcenter, ycenter, cutout_size):
+
+    '''
+    A function that creates a figure showing the
+    provided data, PSF model, and model residuals.
+    '''    
+
+    # Make a figure showing the data, model, and residuals.
+    my_figure_size, my_fontsize = setup_matplotlib('notebook', 1.2)
+    figure, axes = matplotlib.pyplot.subplots(1, 3, figsize=(11.0, 11.0/3))
+    # current_width = 1.2 * matplotlib.pyplot.rcParams["lines.linewidth"]
+    norm = simple_norm(sci_data, 'sqrt', percent=99.8)
+    axes[0].imshow(sci_data, origin='lower', norm=norm, aspect='equal', interpolation='nearest')
+    axes[1].imshow(sci_data - resid, origin='lower', norm=norm, aspect='equal', interpolation='nearest')
+    axes[2].imshow(resid, origin='lower', norm=norm, aspect='equal', interpolation='nearest')
+    axes[0].set_title('Data')
+    axes[1].set_title('Model')
+    axes[2].set_title('Residuals')
+    for x in [0, 1, 2]: 
+        axes[x].set_xlim([xcenter-cutout_size/2, xcenter+cutout_size/2])
+        axes[x].set_ylim([ycenter-cutout_size/2, ycenter+cutout_size/2])
+
+    return figure
+
+
+def download_psf_model(file_path, detector, filter):
+
+    '''
+    Download a PSF model from the WFC3 website and validate the file.
+    '''
+
+    if (detector not in ['WFC3UV', 'WFC3IR', 'ACSWFC']):
+        print('The valid detector options are: WFC3UV, WFC3IR, or ACSWFC')
+
+    psf_name = 'PSFSTD_{}_{}.fits'.format(detector, filter)
+    psf_path = '{}/{}'.format(file_path, psf_name)
+    psf_url = 'https://www.stsci.edu/files/live/sites/www/files/home/hst/instrumentation/wfc3/data-analysis/psf/_documents/'
+
+    # Download the PSF file if it doesn't exist.
```

Comment:
Originally I thought that hst1pass required the PSF models in the same directory as the executable, but it turns out that in completing the file validation check the function was looking for the file rather than the filepath. I've updated this and now the PSF models, both from STScI and those produced by the user, are only saved once in the `psfs_dir`. Thanks!